src/auto-evo/AutoEvoRun.cs

```diff
@@ -322,13 +322,14 @@ protected virtual void GatherInfo(Queue<IRunStep> steps)
                 }
                 else
                 {
-                    steps.Enqueue(new FindBestMutation(autoEvoConfiguration, map, speciesEntry.Key,
+                    steps.Enqueue(new FindBestMutation(autoEvoConfiguration, map, speciesEntry.Key, entry.Value,
```

Comment:
Why is the patch passed through here? This doesn't make sense as the patch the species first appears in is arbitrary (I mean the first patch that appears in the list of patches that a species is in, is not in any order that makes sense to only work on the one patch), it should not be relied on for anything.

Reply:
To log into that particular patch that some species has mutated, and only for that reason. Is there an alternative way to do this by any chance? I don't really understand well this area of the codebase.

Reply:
To do it properly you would have to add a step that runs at the end of auto-evo, and it needs to do a loop of each species checking if it has mutations and each patch they have population in and add the notice that they had a mutation there. That's the only way to do this properly as mutations don't happen in one patch. Also one potentially tricky thing is that you'd also need to handle the species split checking there as the species may have split and some patches no longer have their population.

So in summary I think it's fundamentally flawed approach to have this here, so I suggest you remove the patch parameters you added to the auto-evo steps and instead add an overall step at the end that goes through the results to create mutation events for all species in patches they are in.

Reply:
It may actually be too early if this overall step is ran in `RunStage.Ended` as it happens before GameWorld's time skip and the events logging need to happen after that.

Reply:
Yeah that approach of having a separate method to record the events from RunResults sounds fine.

Also skipping including mutation events in the timeline is probably good as probably most species have mutations pretty often.

Reply:
Alright, sounds good.